include/addon-tools.hpp

```diff
@@ -250,23 +267,20 @@ inline Type* getArrayData(V8_VAR_OBJ obj, int *num = nullptr) {
 	
 	Type *data = nullptr;
 	
-	if (num) {
-		*num = 0;
-	}
-	
-	if ( ! obj->IsArrayBufferView() ) {
+	if (obj->IsArrayBuffer()) {
+		v8::Local<v8::ArrayBuffer> ab = obj.As<v8::ArrayBuffer>();
+		data = reinterpret_cast<Type*>(ab->GetContents().Data());
+		if (num) { *num = ab->ByteLength() / sizeof(Type); }
+	} else if (obj->IsTypedArray()) {
+		v8::Local<v8::TypedArray> ta = obj.As<v8::TypedArray>();
+		data = static_cast<Type*>(ta->Buffer()->GetContents().Data()) + ta->ByteOffset();
```

Comment:
what happens to `+ ta->ByteOffset()` if `Type` = `int`? maybe it will be easier to discard template and always use just `uint8_t` as the return type? otherwise it requires two steps, first get the byte pointer with offset, and then convert to `Type`. I would prefer just using `uint8_t`